compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -1189,36 +1189,30 @@ private BField createFieldWithType(BField field, List<BType> bTypes) {
     }
 
     /**
-     * Get the resultant union or non-union type from a {@code List<BType>}.
+     * Get the resultant type from a {@code List<BType>}.
      *
      * @param bTypes bType list (size > 0)
      * @return {@code BUnionType} if effective members in list is > 1. {@code BType} Otherwise.
      */
-    private BType getResultantUnionOrNonUnionType(List<BType> bTypes) {
-        if (bTypes.size() == 1) {
-            return bTypes.get(0);
-        }
-
+    private BType getResultantType(List<BType> bTypes) {
         LinkedHashSet<BType> bTypeSet = new LinkedHashSet<>(bTypes);
-        List<BType> eBTypes = new ArrayList<>(bTypes.size());
-        addEffectiveMemberTypes(eBTypes, bTypeSet);
+        List<BType> flattenBTypes = new ArrayList<>(bTypes.size());
+        addFlattenMemberTypes(flattenBTypes, bTypeSet);
 
-        return getRepresentativeBroadType(eBTypes);
+        return getRepresentativeBroadType(flattenBTypes);
     }
 
-    private void addEffectiveMemberTypes(List<BType> eBTypes, LinkedHashSet<BType> bTypes) {
+    private void addFlattenMemberTypes(List<BType> flattenBTypes, LinkedHashSet<BType> bTypes) {
```

Comment:
Isn't this now the same as `org.wso2.ballerinalang.compiler.semantics.analyzer.Types#getAllTypes`?